src/torchmetrics/functional/image/lpips.py

```diff
@@ -0,0 +1,421 @@
+
+import torch
+import torchvision
+from packaging import version
+from torch import Tensor, nn
+from torchvision import models as tv
+from typing_extensions import Literal
+
+from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE, _TORCHVISION_GREATER_EQUAL_0_13
+
+_weight_map = {
+    "squeezenet1_1": "SqueezeNet1_1_Weights",
+    "alexnet": "AlexNet_Weights",
+    "vgg16": "VGG16_Weights",
+}
+
+if not _TORCHVISION_AVAILABLE:
+    __doctest_skip__ = ["learned_perceptual_image_patch_similarity"]
+
+
+def _get_net(net: str, pretrained: bool):
+    if _TORCHVISION_GREATER_EQUAL_0_13:
+        if pretrained:
+            pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
+        else:
+            pretrained_features = getattr(tv, net)(weights=None).features
+    else:
+        pretrained_features = getattr(tv, net)(pretrained=pretrained).features
+    return pretrained_features
+
+
+class Squeezenet(torch.nn.Module):
+    """Squeezenet implementation."""
+
+    def __init__(self, requires_grad: int = False, pretrained: int = True):
+        super().__init__()
+        pretrained_features = _get_net("squeezenet1_1", pretrained)
+
+        self.slice1 = torch.nn.Sequential()
+        self.slice2 = torch.nn.Sequential()
+        self.slice3 = torch.nn.Sequential()
+        self.slice4 = torch.nn.Sequential()
+        self.slice5 = torch.nn.Sequential()
+        self.slice6 = torch.nn.Sequential()
+        self.slice7 = torch.nn.Sequential()
+        self.N_slices = 7
+        for x in range(2):
+            self.slice1.add_module(str(x), pretrained_features[x])
+        for x in range(2, 5):
+            self.slice2.add_module(str(x), pretrained_features[x])
+        for x in range(5, 8):
+            self.slice3.add_module(str(x), pretrained_features[x])
+        for x in range(8, 10):
+            self.slice4.add_module(str(x), pretrained_features[x])
+        for x in range(10, 11):
+            self.slice5.add_module(str(x), pretrained_features[x])
+        for x in range(11, 12):
+            self.slice6.add_module(str(x), pretrained_features[x])
+        for x in range(12, 13):
+            self.slice7.add_module(str(x), pretrained_features[x])
+        if not requires_grad:
+            for param in self.parameters():
+                param.requires_grad = False
+
+    def forward(self, x):
+        """Process input."""
+        h = self.slice1(x)
+        h_relu1 = h
+        h = self.slice2(h)
+        h_relu2 = h
+        h = self.slice3(h)
+        h_relu3 = h
+        h = self.slice4(h)
+        h_relu4 = h
+        h = self.slice5(h)
+        h_relu5 = h
+        h = self.slice6(h)
+        h_relu6 = h
+        h = self.slice7(h)
+        h_relu7 = h
+        vgg_outputs = namedtuple("SqueezeOutputs", ["relu1", "relu2", "relu3", "relu4", "relu5", "relu6", "relu7"])
+        out = vgg_outputs(h_relu1, h_relu2, h_relu3, h_relu4, h_relu5, h_relu6, h_relu7)
+
+        return out
```

Comment:
```suggestion

    def forward(self, x):
        """Process input."""
        vgg_outputs = namedtuple("SqueezeOutputs", ["relu1", "relu2", "relu3", "relu4", "relu5", "relu6", "relu7"])
        
        relus = []
        for slice in self.slices:
            x = slice(x)
            relus.append(x)
        return vgg_outputs(*relus)
```

And I'd define vgg_outputs `namedtuple` outside of the method/class.